rascal-vscode-extension/package.json

```diff
@@ -147,6 +147,9 @@
         "scopes": {
           "ambiguity": [
             "invalid"
+          ],
+          "uncategorized": [
+            "meta.embedded"
```

Comment:
VS Code primarily supports `meta.embedded` for TextMate grammars with embedded languages, but as stated in their [documentation](https://code.visualstudio.com/api/language-extensions/syntax-highlight-guide#injection-grammars) (subsection "token types and embedded languages"), it seems to serve our use case, too:

> you can use a `meta.embedded.*` scope to reset VS Code's marking of tokens

I couldn't find a more suitable scope, neither [here](https://macromates.com/manual/en/language_grammars), nor [here](https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide#semantic-token-provider), nor [there](https://stackoverflow.com/questions/76574973/what-textmate-scope-is-used-for-normal-code-text-in-vs-code)...

Reply:
Can we also take a look in the source code of the vs code rendering which it uses as fallback/default?

since embedded can also be used for cases like html inside of typescript. (jsx)

Reply:
Each theme defines general colors and token colors. For instance ([Dark](https://github.com/microsoft/vscode/blob/main/extensions/theme-defaults/themes/dark_vs.json)):

```
{
	"$schema": "vscode://schemas/color-theme",
	"name": "Dark (Visual Studio)",
	"colors": {
		"checkbox.border": "#6B6B6B",
		"editor.background": "#1E1E1E",
		"editor.foreground": "#D4D4D4",
		...
	},
	"tokenColors": [
    ...
		{
			"scope": "comment",
			"settings": {
				"foreground": "#6A9955"
			}
		},
		...
	]
}
```

In this theme, the only standard token colors that have the same `foreground` property as the general `editor.foreground` property are:

  - `meta.embedded`
  - `meta.template.expressions`
  - `keyword.operator`

Of these, combined with the quote in my previous comment, I think `meta.embedded` is the most suitable for us.

Reply:
`editor.foreground` is not a valid scope I'm guessing?

and there is no -1 token we can pass in?

Reply:
I've just found:

```
const enum SemanticTokensProviderStylingConstants {
	NO_STYLING = 0b01111111111111111111111111111111
}
```

in the source code of VS, which is triggered if we send an token index that is not in the list of registered tokens. Have you tried what happens if we send an invalid token number? looking at the code, it might just be ignored in that case.

Otherwise, agreed, lets use `meta.embedded`.

Reply:
I can't seem to get `""` to work. Briefly discussed and decided to keep `meta.embedded` for now.